src/main/com/yetanalytics/persephone.cljc

```diff
@@ -531,28 +534,35 @@
 
 ;; Statement Pattern Matching ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(defn- cond-println
+  "Println only if `print?` is `true`."
+  [print? expr]
+  (when print? (println expr)))
```

Comment:
This kind of seems like over-refactoring to me, you're taking something that a Clojure dev can read instantly and making it something they have to go look up...

Reply:
I realize that threading print control in is a challenge, as would be threading error messages out.

Reply:
Yeah come to think of it, this `print?` arg I'm passing around is like a poor man's logger. It would feel strange to put logging in a library though.

Reply:
Yeah the timestamps and all that don't really jive with what you're doing here. But it does seem like threading the print arg around is kind of leading to complexity. Hard to say what the best thing to do there is.

Reply:
I admit having `print?` as a kwarg is particularly annoying since then you can't have variable arities (which is why there is no binary version of `match-statement[-batch]`)